tests/baselines/reference/templateStringInModuleName.errors.txt

```diff
@@ -1,32 +1,26 @@
 tests/cases/conformance/es6/templates/templateStringInModuleName.ts(1,1): error TS2304: Cannot find name 'declare'.
-tests/cases/conformance/es6/templates/templateStringInModuleName.ts(1,9): error TS1005: ';' expected.
 tests/cases/conformance/es6/templates/templateStringInModuleName.ts(1,9): error TS2580: Cannot find name 'module'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node`.
-tests/cases/conformance/es6/templates/templateStringInModuleName.ts(1,21): error TS1005: ';' expected.
+tests/cases/conformance/es6/templates/templateStringInModuleName.ts(1,16): error TS1443: Module declaration names may only use ' or " quoted strings.
 tests/cases/conformance/es6/templates/templateStringInModuleName.ts(4,1): error TS2304: Cannot find name 'declare'.
-tests/cases/conformance/es6/templates/templateStringInModuleName.ts(4,9): error TS1005: ';' expected.
 tests/cases/conformance/es6/templates/templateStringInModuleName.ts(4,9): error TS2580: Cannot find name 'module'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node`.
-tests/cases/conformance/es6/templates/templateStringInModuleName.ts(4,24): error TS1005: ';' expected.
+tests/cases/conformance/es6/templates/templateStringInModuleName.ts(4,16): error TS1443: Module declaration names may only use ' or " quoted strings.
 
 
-==== tests/cases/conformance/es6/templates/templateStringInModuleName.ts (8 errors) ====
+==== tests/cases/conformance/es6/templates/templateStringInModuleName.ts (6 errors) ====
     declare module `M1` {
     ~~~~~~~
 !!! error TS2304: Cannot find name 'declare'.
             ~~~~~~
-!!! error TS1005: ';' expected.
-            ~~~~~~
 !!! error TS2580: Cannot find name 'module'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node`.

```

Comment:
This becomes unfortunate because you will now possibly get a semantic error. Ideally, we would just allow no-substitution template strings in these positions and gracefully handle them later in the compiler.

I'm not sure if it's that big of a problem though. I'd hold off on any changes here.

Reply:
In hindsight, I’m actually surprised this wasn’t one of the places that IllusionMH made work with NoSubstitutionTemplateLiterals. I thought we hit basically everything that didn’t conflict with the ES spec. Why even bother issuing a grammar error? But that’s a discussion for another PR 🙃

Reply:
A few clarification requests on this thread:
* _"possibly get a semantic error"_ -- is giving a `parseErrorAt` with the `Module declaration names may only use ' or " quoted strings` not going to stop that? I'm under the impression this PR only changes which diagnostic is emitted, not the phase.
* I'm inferring _"no-substitution template strings"_ to be template strings that don't have any `${...}`s in them?
* By _"hold off on any changes here"_ do you mean I should leave the PR as-is, or revert the logic back to the way it was before this PR?

Reply:
Ping @andrewbranch @DanielRosenwasser 👋 , I think this PR is ready for re-review pending my questions in this thread? 🙏

Reply:
I’m also not sure what Daniel meant on this one. This shouldn’t change semantic errors since the parse tree isn’t changing, as far as I can tell. My approval on the PR still stands, just want to make sure Daniel is happy with the changes to address his feedback.

Reply:
Ping, @DanielRosenwasser? 🙏

Reply:
I meant leave the PR as-is, sorry!

> Module declaration names may only use ' or " quoted strings.

Don't know if I caught this diagnostic - I think that should be sufficient!